src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -141,19 +144,30 @@ public with sharing class BDI_AdditionalObjectService {
 
                             //If the predecessor doesn't exist, then update the Data import object with an error 
                             //and continue with the next dataImport.
-                            if (!predecessorExists(objWrap,predecessor)) {
-                                continue;
+                            if (!dataImportService.isDryRun) {
```

Comment:
The reason I put this isDryRun check in here is because the predecessor check will always fail during dry run since the predecessor objects are not created during dry run.